pytorch_lightning/loggers/comet.py

```diff
@@ -160,9 +173,13 @@ def log_metrics(self, metrics, step=None):
 
         self.experiment.log_metrics(metrics, step=step)
 
+    def reset_experiment(self):
+        self._experiment = None
+
     @rank_zero_only
     def finalize(self, status):
         self.experiment.end()
+        self.reset_experiment()
```

Comment:
What is the need for setting `self._experiment` to None?

Reply:
When calling `self.experiment.end()`, that `experiment` won't log any more data to Comet (that was issue #760 about). That's why, if you need to log any more data you need to create an `ExistingCometExperiment`. For example, to log data when testing your model after training, because when training is finalized `CometLogger.finalize` is called. 

With this PR, this happens automatically in the `CometLogger.experiment` property, when `self._experiment` is set to `None`.

Tell me if anything is not clear yet :)

Reply:
I think it would be nice to write this your explanation also to code as a comment for future... :robot: